### PR TITLE
Add cloud config stages for installation/upgrade/deploy

### LINKED ISSUE #154: Installation/Upgrade as cloud-config stages
**Is your feature request related to a problem? Please describe.**
Customization during install and upgrade phase are not possible, e.g. even displaying a simple informative message

**Describe the solution you'd like**
A stage, like we do for `boot`, `network`, ... but for `installation` and `upgrade`. In this way we allow to customize those aspects of the derivatives as well with a generic approach (and we keep everything in cloud-config)

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by mudler**
Adds after-install, after-deploy, after-upgrade, before-install,
before-deploy, before-upgrade as cloud-config stages. This means that
now cloud configuration files can hook into those stages as well for
further customization.

Fixes #154

Signed-off-by: Ettore Di Giacinto <edigiacinto@suse.com>

### CODE REVIEW

**Review by Itxaka — APPROVED** [review]
nice!

**Comment by Itxaka** [comment]
should we also add some before reset? Just thinking that it may be useful for something down the line

**Comment by mudler** [comment]
good point! added :+1:

**Review by Itxaka — APPROVED** [review]
was already green before, no point in waiting for tests to merge

**Pull request merged**
